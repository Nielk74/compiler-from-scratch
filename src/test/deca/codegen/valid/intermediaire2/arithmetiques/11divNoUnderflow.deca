// Description:
//    tests de programme correct codegen : 
//    diff√©rentes divisions (inclut MIN en 32b 1.0p-149) qui lancent pas l'exception underflow
//
// Resultats:
//  0x1.c92492p-1
//  0x1.1eb852p+0
//  0x1.4p+2
//  0x1p-149
//  0x0p+0
//
// Historique:
//    cree le 15/06/2022
{
    int a = 5;
    float b = 5.6;
    float c;
    
    c = a/b;
    printlnx(c);
    c = b/a;
    printlnx(c);
    c = b/c;
    printlnx(c);
    c = 2;
    b = 0x1.0p-148;
    c = b/c;
    printlnx(c);
    b = 0;
    c = b/c;
    printlnx(c);
}