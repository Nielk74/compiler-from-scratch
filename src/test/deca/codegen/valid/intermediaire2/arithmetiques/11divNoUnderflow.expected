0x1.c92492p-1
0x1.1eb852p+0
0x1.4p+2
0x1p-149
0x0p+0
